nautobot/extras/tests/test_dynamicgroups.py

```diff
@@ -629,6 +630,18 @@ def test_generate_query_for_filter(self):
         serial_qs = Device.objects.filter(serial__iexact=solo_value)
         self.assertQuerySetEqual(solo_qs, serial_qs)
 
+        # Test that a nested field_name w/ `generate_query` works as expected. This is explicitly to
+        # test a regression w/ nested slug-related values such as `DeviceFilterSet.region` which
+        # filters on `site__region`.
+        parent_region = Region.objects.filter(children__isnull=False).first()
+        nested_value = [parent_region.slug]
+        group.set_filter({"region": nested_value})
+        group.validated_save()
+        nested_query = group.generate_query_for_filter(filter_field=fs.filters["region"], value=nested_value)
+        nested_qs = queryset.filter(nested_query)
+        region_qs = Device.objects.filter(site__region__slug__in=nested_value)
```

Comment:
If it were empty, this wouldn't even be relevant since `generate_query_for_filter`, but all the same it's more critical that we are making sure the filterset generated from the slug as an argument results in the same filtered queryset, and more importantly that the nested filter expression `site__region` is automatically used to get the related model name without failing. 

I revised the comment to say this.